### PR TITLE
Switch branch: Numerous improvements, see description

### PR DESCRIPTION
**Pull request by rsn8887**
This PR
- adds error handling on opening joystick and game controller
- removes mman.h dependency altogether for Switch platform
- adds link to MVG profile to readme-switch
- remove hard-coded window size on Switch
- maps character deletion in menu to top controller button (button X on Switch)
- adds information to readme-switch.md how to delete character using controller button X press
- enables network and stdio on Switch

### CODE REVIEW

File: Source/init.cpp [hunk 1]
```diff
@@ -157,6 +157,7 @@ void init_create_window(int nCmdShow)
 	wcex.hIconSm = (HICON)LoadImage(ghInst, MAKEINTRESOURCE(IDI_ICON1), IMAGE_ICON, 16, 16, LR_DEFAULTCOLOR);
 	if (!RegisterClassEx(&wcex))
 		app_fatal("Unable to register window class");
+#endif
```
Comment (AJenbo) [inline Source/init.cpp]: Wait is this needed?

Comment (rsn8887) [reply Source/init.cpp]: Probably not, but cannot test atm. Will test later.

Comment (rsn8887) [reply Source/init.cpp]: Not needed, you were right.

Fixed in https://github.com/diasurgical/devilutionX/pull/201/commits/00ee02c4d39f9bdf338ff9b305bf13666a0afb83

**Comment by AJenbo** [comment]
Could you enable Travis and circle ci access on your account so that your PR is tested before merge?

**Comment by rsn8887** [comment]
I don’t know how to do that and never had to do that for any other repo. Do you have some suggestion where to look for this option?

**Comment by rsn8887** [comment]
I suspect maybe an option in CircleCI that you can set might fix this https://circleci.com/docs/2.0/oss/

It is called advanced settings/build forked pull request

**Comment by AJenbo** [comment]
Arh that was probably it, thanks for looking it up.

**Comment by rsn8887** [comment]
Tested and ready to merge :)

File: SourceX/miniwin/misc_msg.cpp [hunk 2]
```diff
@@ -5,6 +5,10 @@
 #include "stubs.h"
 #include <math.h>
 #include "../../touch/touch.h"
+#ifdef SWITCH
+	#include "ddraw.h"
```
Comment (AJenbo) [inline SourceX/miniwin/misc_msg.cpp]: I this DirectDraw? Why would it be needed?

**Comment by AJenbo** [comment]
@rsn8887 PR looks good, the only thing I'm wondering about is why `ddraw.h` is required now?

File: SourceX/miniwin/misc_msg.cpp (hunk 2 above)
Comment (rsn8887) [reply SourceX/miniwin/misc_msg.cpp]: It is the DD wrapper yes. I only needed `window` and `renderer` references from it. I fixed it just now, using externals instead.

**Comment by rsn8887** [comment]
ddraw was just used to gain access to `window` and `renderer` for the automatic resolution switching when docking/undocking the switch. I agree I shouldn't have included the full thing. I changed it here, using externals:
https://github.com/diasurgical/devilutionX/blob/4b4c78668d716c78001f38dcc8949bafb21139ac/SourceX/miniwin/misc_msg.cpp#L45

**Comment by AJenbo** [comment]
Looks good, nice improvements :+1:

**Comment by rsn8887** [comment]
Thanks for merging. You are right it is better with the externals.

**Pull request merged**